### ISSUE TITLE
Very bad performance of TS and JS Language Features

### ISSUE THREAD

**Issue by sarneeh** [microsoft/vscode#73624]
I struggle with a very bad performance (I mean mostly intellisense) in VS Code recently so I started to dig down to find the root cause of the problem.

What I found was:
- performance issues go away when I set `javascript.validate.enable` to `false`
- performance issues go away when I rename my `jsconfig.json` to `tsconfig.json`

What's weird is that the problems occur only when using the `jsconfig.json` configuration file. Here's how it looks like:

```js
{
  "compilerOptions": {
    "baseUrl": "./",
    "target": "es5",
    "paths": {
      "@page/*": ["src/page/*"],
      "@panel/*": ["src/panel/*"],
      "@shared/*": ["src/shared/*"]
    },
    "experimentalDecorators": true
  },
  "exclude": ["node_modules", ".next"]
}
```

When I rename the same file to `tsconfig.json` - the problems disappear.

The difference I in both approaches was visible in the TS Server Log - using `jsconfig.json`, the TS Server parsed ~400 more files than using the `tsconfig.json` version.

I attached the logs for specific configurations.

[jsconfig.json](https://github.com/microsoft/vscode/files/3169525/tsserver.log)
[tsconfig.json](https://github.com/microsoft/vscode/files/3169526/tsserver.log)

**Comment by icyJoseph** [comment]
I have a similar issue. Starting vscode is painful experience, because of TS and JS Language server being started. This seems to be directly related to the amount of packages installed on the project. 

However with the, TypeScript and JavaScript Language Features builtin extension, disabled I have zero issues. 

<details>
<summary>vscode</summary>
<br>
Version: 1.33.1
Commit: 51b0b28134d51361cf996d2f0a1c698247aeabd8
Date: 2019-04-11T08:20:22.771Z
OS: Linux x64 5.0.9-050009-generic
</details>

<details>
<summary>System</summary>
<br>
OS: elementary OS 5.0 Juno x86_64 
Host: Blade Stealth 1.04 
Kernel: 5.0.9-050009-generic 
Uptime: 1 day, 2 hours, 43 mins 
CPU: Intel i7-6500U (4) @ 3.100GHz 
GPU: Intel Skylake GT2 [HD Graphics 520] 
Memory: 3921MiB / 7742MiB 
</details>

**Comment by mjbvz** [comment]
@sarneeh Does your project only contain javascript files?

**Comment by sarneeh** [comment]
@mjbvz Yes, just javascript files. It didn't make sense to me to change the file to `tsconfig` but I've tried anyways and... It helped a little :smile:
